### PR TITLE
APID-39: adds support for passing in a notification URL and sending status updates to this URL

### PR DESCRIPTION
**Pull request by worthydolt**

### CODE REVIEW

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/OutboundMessageRequest.scala [hunk 1]
```diff
@@ -22,7 +22,8 @@ case class MessageRequest(wsdlUrl: String,
                           wsdlOperation: String,
                           messageBody: String,
                           addressing: Option[Addressing] = None,
-                          confirmationOfDelivery: Boolean)
+                          confirmationOfDelivery: Boolean,
+                          notificationUrl: Option[String] = None)
```
Comment (lburgos) [inline app/uk/gov/hmrc/apiplatformoutboundsoap/models/OutboundMessageRequest.scala:24]: We should add this to the readme

File: test/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundServiceSpec.scala [hunk 2]
```diff
@@ -316,6 +322,40 @@ class OutboundServiceSpec extends AnyWordSpec with Matchers with GuiceOneAppPerS
...
+      when(appConfigMock.parallelism).thenReturn(2)
+      when(appConfigMock.retryInterval).thenReturn(Duration("1s"))
+      when(outboundConnectorMock.postMessage(*)).thenReturn(successful(OK))
+      when(outboundMessageRepositoryMock.updateStatus(*, *)).thenReturn(successful(Some(sentMessageForNotification)))
+      when(outboundMessageRepositoryMock.retrieveMessagesForRetry).
+        thenReturn(fromFutureSource(successful(fromIterator(() => Seq(retryingMessage).toIterator))))
+
+      await(underTest.retryMessages)
+
+      verify(notificationCallbackConnector).sendNotification(refEq(sentMessageForNotification))(*)
```
Comment (lburgos) [inline test/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundServiceSpec.scala]: We could add a test where this call fails but the retrying message is still set to SENT

File: app/uk/gov/hmrc/apiplatformoutboundsoap/models/OutboundMessageRequest.scala (hunk 1 above)
Comment (worthydolt) [reply app/uk/gov/hmrc/apiplatformoutboundsoap/models/OutboundMessageRequest.scala:24]: Done

File: test/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundServiceSpec.scala (hunk 2 above)
Comment (worthydolt) [reply test/uk/gov/hmrc/apiplatformoutboundsoap/services/OutboundServiceSpec.scala]: Tested in a new scenario added to this spec class. Also renamed one mock to make its mockness clearer

**Review by lburgos — APPROVED** [review]

**Pull request merged**